code-walktrhough.md

```diff
@@ -0,0 +1,215 @@
+# Walkthrough of the RISC Zero Foundry Template.
+
+This walkthrough covers the Even Number example provided with the RISC Zero Foundry Template.
+
+It is built with [Foundry](https://github.com/foundry-rs/foundry), a powerful set of tools for Ethereum smart contract development, including testing and deployment. The template leverages the RISC Zero zkVM to ensure the integrity of computations on the Ethereum blockchain. It includes a smart contract, guest code for off-chain computations, a publisher (CLI) application for interacting with Ethereum, and tests to ensure everything works as expected.
+
+## Guest code overview
+
+The guest code runs off-chain within the RISC Zero zkVM. It performs the actual computation to check if a given number is even, generating a proof of this computation. This code forms the basis for the off-chain computation part of the Ethereum ZK-Coprocessor.
+
+### Guest code walkthrough:
+
+```rust
```

Comment:
Are we already (and/or is it practical to) testing code snippets like this in CI?